common/src/commonMain/kotlin/dev/msfjarvis/claw/common/comments/CommentEntry.kt

```diff
@@ -0,0 +1,82 @@
+package dev.msfjarvis.claw.common.comments
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Divider
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import dev.msfjarvis.claw.common.posts.PostDetails
+import dev.msfjarvis.claw.common.posts.SubmitterName
+import dev.msfjarvis.claw.common.posts.toDbModel
+import dev.msfjarvis.claw.model.Comment
+import dev.msfjarvis.claw.model.LobstersPostDetails
+
+@Composable
+fun CommentsHeader(
+  postDetails: LobstersPostDetails,
+) {
+  Surface {
+    Column(
+      modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp).fillMaxWidth(),
+      verticalArrangement = Arrangement.spacedBy(4.dp),
+    ) {
+      PostDetails(
+        post = postDetails.toDbModel(),
+      )
+    }
+  }
+}
+
+@Composable
+fun CommentEntry(
+  comment: Comment,
+  renderMarkdown: @Composable (comment: String, modifier: Modifier) -> Unit,
+) {
+  val indentLevel = comment.indentLevel.toInt() - 1
+  val startPadding = ((10 * indentLevel) + 16).dp
+
+  Divider(color = Color.Gray.copy(0.4f))
+  Row(modifier = Modifier.height(IntrinsicSize.Min)) {
+    CommentTreeColors(indentLevel = indentLevel)
+    Column(
+      modifier = Modifier.padding(start = startPadding, end = 8.dp, top = 4.dp, bottom = 4.dp)
+    ) {
+      SubmitterName(
+        text = "Submitted by ${comment.user.username}",
+        avatarUrl = "https://lobste.rs/${comment.user.avatarUrl}",
+        contentDescription = "Submitted by ${comment.user.username}",
+      )
+      renderMarkdown(comment.comment, Modifier.padding(top = 8.dp))
+    }
+  }
+}
+
+@Composable
+private fun CommentTreeColors(
+  indentLevel: Int,
+  modifier: Modifier = Modifier,
+) {
+  Box(modifier = modifier) {
+    for (level in 1..indentLevel) {
+      Box(
+        modifier =
+          Modifier.padding(start = (level * 12).dp)
```

Comment:
You probably don't want a linear scale here. Third/Fourth level indentation looks really bad with this. You can probably do 12dp for the first level and then 4/6dp for each subsequent level.